Correctly copy & paste data with context menu, and handle number notion that is based on locale

Michota opened the pull request:
## Context menu copy-pasting
- Fix issue #282 
- New function getRowsFromClipboard (asynchronous):
  - returns Compatible<Cell>[], which can be later used in pasteData function.
### Parse number with respecting locale number notion
Respect the locale settings for user-supplied numerical data, whether pasted or entered manually.
- New function parseLocaleNumber
  - Locale can be provided manually (as string). If this does not happen, the function will get it from the browser itself.
  - returns number.
```ts
function parseLocaleNumber(stringNumber: string, locale = getNavigatorLanguage()): number
```
```tsx
parseLocaleNumber("123456,78") // returns 123456.78
parseLocaleNumber("1 234 56,78", "pl-PL") // returns 123456.78
parseLocaleNumber("1,234,56.78", "en-US") // returns 123456.78

Michota committed:
chore: change configs settings

- TypeScript config: noEmitOnError: true, for dev purposes
- Prettier: maxWidth: 120

Michota committed:
fix: parse numbers on pasting based on user's locale

Michota committed:
fix: parseLocaleNumber returns correct results

- may not work for arabic number notations

Michota committed:
fix: re-write parseLocaleNumber

- create function that will get decimal and thousand separators
- replace these separators in strings passed as an stringNumber argument

Michota committed:
fix: re-write parseLocaleNumber

- create helper-function that will get decimal and thousand separators
- replace these separators in strings passed as an stringNumber argument
- replace all characters (like "USD", or "$", etc.) before and after numbers

Michota committed:
feat:getLocaleSeparators

fix after-merge conflict

Michota committed:
chore: temporary remove test before push to develop

Michota committed:
chore: temporary remove test before push to develop

fix: re-write parseLocaleNumber

- create helper-function that will get decimal and thousand separators
- replace these separators in strings passed as an stringNumber argument
- replace all characters (like "USD", or "$", etc.) before and after numbers

Michota committed:
Merge branch 'v4-fix/context-menu-paste' of https://github.com/silevis/reactgrid into v4-fix/context-menu-paste

Michota committed:
Merge branch 'v4-fix/context-menu-paste' of https://github.com/silevis/reactgrid into v4-fix/context-menu-paste

Michota committed:
Merge branch 'v4-fix/context-menu-paste' of https://github.com/silevis/reactgrid into v4-fix/context-menu-paste

Michota committed:
fix: paste numbers to cell value correctly

- fix issue no. #282

Michota committed:
fix: paste numbers to cell value correctly

- fix issue no. #282

Michota committed:
Merge branch 'v4-fix/context-menu-paste' of https://github.com/silevis/reactgrid into v4-fix/context-menu-paste

Michota committed:
Merge branch 'v4-fix/context-menu-paste' of https://github.com/silevis/reactgrid into v4-fix/context-menu-paste

Michota committed:
Merge branch 'v4-fix/context-menu-paste' of https://github.com/silevis/reactgrid into v4-fix/context-menu-paste

Michota committed:
fix: correctly handle pasting with context menu

feat: getRowsFromClipboard function

- fixes issue #282

DLowHP commented on package.json:
```diff
@@ -80,7 +80,6 @@
   },
   "husky": {
     "hooks": {
-      "pre-push": "if git-branch-is develop; then npx run-s eslint run:all:tests; fi"
```
I believe this was commited by mistake as I think we should keep this line.

Michota replied:
Yeah, You are right. But for some reason i can't push if this hook is there, even if i try to push it to non-develop branch...

DLowHP replied:
Will have to look into this some time... I'll add this to the backlog!

DLowHP commented on src/lib/Functions/getNavigatorLanguage.ts:
```diff
@@ -0,0 +1,8 @@
+  // navigator.userLanguage || 
+  navigator.language ||
+  // navigator.browserLanguage ||
+  "en-US";
```
Does this need to be so complicated? I think that `navigator.language` should be sufficient, but I might be wrong?

Michota replied:
Oh, you are right, `navigator.language` is basically the same as `navigator.languages[0]`

DLowHP commented on tsconfig.json:
```diff
@@ -18,6 +18,7 @@
     "jsx": "react-jsx",
     "noImplicitAny": true,
     "noEmit": true,
+    "noEmitOnError": false,
```
Is this _explicitly_ needed? [The docs](https://www.typescriptlang.org/tsconfig#noEmitOnError:~:text=This%20defaults%20to%20false%2C%20making%20it%20easier%20to%20work%20with%20TypeScript%20in%20a%20watch%2Dlike%20environment%20where%20you%20may%20want%20to%20see%20results%20of%20changes%20to%20your%20code%20in%20another%20environment%20before%20making%20sure%20all%20errors%20are%20resolved.) say that this is `false` by default.

DLowHP committed:
getRowsFromClipboard: readability and security improvements, added types
Added missing types, error handles, moved parts of the logic to separate functions, renamed functions and variables to better indicate their purpose

DLowHP committed:
parseLocaleNumber: added disclaimer; formatting

Michota committed:
style: remove unnecessary code

Michota committed:
chore: bring back pre-push hook

Michota committed:
Merge branch 'develop' into v4-fix/context-menu-paste
